### PR TITLE
Tests for finding stuff

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/arrayops.jl [hunk 1]
```diff
@@ -381,6 +381,14 @@ end
 
     @test findall(in([1, 2]), 2) == [1]
     @test findall(in([1, 2]), 3) == []
+
+    @test sort(findall(Dict(1=>false, 2=>true, 3=>true))) == [2, 3]
+
+    @test findall(true) == [1]
+    @test findall(false) == Vector{Int}[]
```
Comment (fredrikekre) [inline test/arrayops.jl]: I guess you meant
```suggestion
    @test findall(false) == Int[]
```

Kinda weird that
```
julia> Int[] == Vector{Int}[]
true
```
but I guess all elements are equal so...

File: test/arrayops.jl [hunk 2]
```diff
@@ -381,6 +381,14 @@ end
...
+
+    @test findall(isodd, 1) == [1]
+    @test findall(isodd, 2) == Vector{Int}[]
```
Comment (fredrikekre) [inline test/arrayops.jl]:
```suggestion
    @test findall(isodd, 2) == Int[]
```

File: test/abstractarray.jl [hunk 3]
```diff
@@ -867,6 +867,8 @@ for A in (rand(2), rand(2,3))
 
     @test keytype(A) == eltype(keys(A))
```
Comment (fredrikekre) [inline test/abstractarray.jl]:
```suggestion
     @test keytype(A) == keytype(typeof(A)) == eltype(keys(A))
```

File: test/abstractarray.jl [hunk 4]
```diff
@@ -867,6 +867,8 @@ for A in (rand(2), rand(2,3))
...
     @test valtype(A) == eltype(values(A))
```
Comment (fredrikekre) [inline test/abstractarray.jl]:
```suggestion
     @test valtype(A) == valtype(typeof(A)) == eltype(values(A))
```

File: test/abstractarray.jl [hunk 5]
```diff
@@ -867,6 +867,8 @@ for A in (rand(2), rand(2,3))
...
+    @test keytype(typeof(A)) == eltype(keys(A))
```
Comment (fredrikekre) [inline test/abstractarray.jl]:
```suggestion
```

File: test/abstractarray.jl [hunk 6]
```diff
@@ -867,6 +867,8 @@ for A in (rand(2), rand(2,3))
...
+    @test valtype(typeof(A)) == eltype(values(A))
```
Comment (fredrikekre) [inline test/abstractarray.jl]:
```suggestion
```

**Comment by kshyatt** [comment]
I'll squash and merge these guys if tests look ok

**Comment by kshyatt** [comment]
@fredrikekre thanks for the feedback! good to go, you think?

**Pull request merged**